gdal/apps/gdal_grid_lib.cpp

```diff
@@ -768,6 +772,12 @@ GDALDatasetH GDALGrid( const char *pszDest, GDALDatasetH hSrcDataset,
         return nullptr;
     }
 
+    if ( (psOptions->nXSize || psOptions->nYSize) && (psOptions->dfXRes || psOptions->dfYRes) ) {
```

Comment:
```suggestion
    if ( (psOptions->nXSize != 0 || psOptions->nYSize != 0) && (psOptions->dfXRes != 0 || psOptions->dfYRes != 0) ) {
```

Reply:
I think that made warnings too. But where do you see them? My local compiler doesn't show me any warnings.
Sometimes I think too complicated to find the solution. I really appreciate your work, thank you for this bug review.

Reply:
This was in the build log of https://travis-ci.com/github/OSGeo/gdal/jobs/371646974  . If you look for the string "error: " you get on the lines with the warnings/errors.
They depend on compiler types & versions, so you can't catch everything locally.

Reply:
Ahh perfect. Thank you very much.